codegen/src/types/type_def.rs

```diff
@@ -70,13 +71,16 @@ impl TypeDefGen {
                 let docs = should_gen_docs.then_some(&*ty.docs).unwrap_or_default();
                 let composite_def = CompositeDef::struct_def(
                     ty,
+                    types_mod_ident,
+                    &type_name,
                     &type_name,
```

Comment:
So we pass the type_name as the `variant_name` parameter here as well, but it is not really used for a normal Composite from the types registry, right?